src/json_io.hh

```diff
@@ -35,11 +35,244 @@
  */
 #include "json.hpp"
 
+#include <Eigen/Dense>
 #include <fstream>
 
 // For convenience
 using json = nlohmann::json;
 
+/**
+ * Utilities to convert json to Eigen types and the oposite
+ */
+namespace Eigen {
+  namespace internal {
+
+    template <int _Rows, int _Cols>
+    struct ResizeEigen {
```

Comment:
ResizeEigen is called if it is a fixed size matrix and it corresponds to a no op. If it was throwing an error if executed then some checks should be done before hand but that would defeat the purpose of this construct.

Reply:
...what? Then I don't understand the purpose of this construct. Where is the check that ensures this doesn't get called with the wrong sizes on fixed-size matrices?

Reply:
there are no checks since it is all templated

Reply:
Wait, but that doesn't stop someone from calling `apply_mat()` with the wrong sizes, even if it is templated. For instance, you could call `ResizeEigen<3, 3>::apply_mat(4, 4)` which, by Eigen rules, shouldn't even compile - but here I think it does, and just silently discards the sizes (the 4, 4) passed in. That can easily mask bugs later on.

Reply:
Ugh, I'm not sure why the automated build breaks; it's just fine on my machine. One option could be to just call the resize and let Eigen complain if the sizes differ...

Reply:
Is this solved?

Reply:
Not quite, not for the case of matrices where both the row and the column size are fixed. (it was sneakily reverted in the merge commit a0aac28...) But for the other two cases, it will call Eigen's resize directly which raises (I think) a run-time error; checking at compile time seems to be either hard or impossible.

Reply:
of course I reverted these asserts since they cause failures that are not warranted. Please check your modifications before committing even if you feel that they should work.